std/algorithm/iteration.d

```diff
@@ -4280,3 +4284,77 @@ unittest
          [1, 2, 0],
          [2, 1, 0]]));
 }
+
+/**
+Finds the mean (colloquially known as the average) of a range.
+Is $(D O(r.length)).
+
+Params:
+    r = a input range
+
+Returns:
+    the mean of r
+*/
+auto mean(R)(R r) if (isInputRange!R && !isInfinite!R) {
+    import std.typecons : tuple;
+
+    immutable auto pair = reduce!((a, b) => tuple(a[0] + 1, a[1] + b))(tuple(0u, 0.0), r);
```

Comment:
This is not going to work well for reals, longs, any ints where the sum overflows, etc
Should probably use `sum`.
Wastes a lot of time re-calculating the length if the range has it available.